src/main.cpp

```diff
@@ -897,332 +929,347 @@ std::string url_encode(const std::string& value) {
+        std::string nameStr = fmt::format("cmd /k cd \"{}\\gdutils\" && \"gd_protocol.bat\" %1 && exit", dirs::getGameDir());
+
+        std::wstring fullPath = std::wstring(buffer.data());
+        size_t lastBackslashPos = fullPath.find_last_of(L"\\");
+        std::wstring exeName;
+        if (lastBackslashPos != std::wstring::npos) {
+            exeName = fullPath.substr(lastBackslashPos + 1);
+        }
+
+        // Create the key for the custom protocol in user registry
+        HKEY hCurrentUserKey;
+        if (RegOpenKeyEx(HKEY_CURRENT_USER, "Software\\Classes", 0, KEY_SET_VALUE, &hCurrentUserKey) == ERROR_SUCCESS) {
+            HKEY hKey;
+            if (RegCreateKeyEx(hCurrentUserKey, "gdutils", 0, NULL, 0, KEY_SET_VALUE, NULL, &hKey, NULL) == ERROR_SUCCESS) {
+                // Define the default value for the protocol key
+                if (RegSetValueEx(hKey, NULL, 0, REG_SZ, (BYTE*)"URL:gdutils Protocol", sizeof("URL:gdutils Protocol")) == ERROR_SUCCESS) {
+                    // Create the key "shell\open\command" for specify the exe path
+                    HKEY hCommandKey;
+                    if (RegCreateKeyEx(hKey, "shell\\open\\command", 0, NULL, 0, KEY_SET_VALUE, NULL, &hCommandKey, NULL) == ERROR_SUCCESS) {
+                        if (RegSetValueExA(hCommandKey, NULL, 0, REG_SZ, (const BYTE*)nameStr.c_str(), nameStr.length() + 1) == ERROR_SUCCESS) {
+                            log::info("Le protocole a été enregistré avec succès pour cet utilisateur.");
+                        } else {
+                            log::error("Erreur lors de la définition de la valeur du chemin de l'exécutable.");
+                        }
+                        RegCloseKey(hCommandKey);
                     } else {
-                        log::error("Erreur lors de la définition de la valeur du chemin de l'exécutable.");
+                        log::error("Erreur lors de la création de la clé shell\\open\\command.");
                     }
-                    RegCloseKey(hCommandKey);
+
+                    RegSetValueEx(hKey, "URL Protocol", 0, REG_SZ, NULL, NULL);
                 } else {
-                    log::error("Erreur lors de la création de la clé shell\\open\\command.");
+                    log::error("Erreur lors de la définition de la valeur par défaut du protocole.");
                 }
-
-                RegSetValueEx(hKey, "URL Protocol", 0, REG_SZ, NULL, NULL);
+                RegCloseKey(hKey);
             } else {
-                log::error("Erreur lors de la définition de la valeur par défaut du protocole.");
+                log::error("Erreur lors de la création de la clé du protocole.");
             }
-            RegCloseKey(hKey);
+            RegCloseKey(hCurrentUserKey);
         } else {
-            log::error("Erreur lors de la création de la clé du protocole.");
+            log::error("Erreur lors de l'ouverture de la clé du registre utilisateur.");
```

Comment:
I find that Jetbrains did more than just generate files. It also reworded some of the logs.

Reply:
i thought i had that turned off, sorry about that.